### PR TITLE
A sandboxed iframe can communicate back to its originating site.

### PR DESCRIPTION
**Pull request by cyril-sf**
Use `allowSameOrigin` to add the `allow-same-origin` flag on the `sandbox` attribute of an iframe.

``` js
oasis.configure('allowSameOrigin', true);
```

Also put back the security for same origin for IE8/9.

Enforcing a different origin between the container and the iframe prevents from creating a sandbox using the default Oasis URL on browsers that don't support the `sandbox` attribute.

### CODE REVIEW

**Comment by cyril-sf** [comment]
@hjdivad good for review. it passes tests https://travis-ci.org/tildeio/oasis.js/builds/12295080

File: test/fixtures/same_origin_parent.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    _addEventListener(window, "message", function (event) {
+      if( event.data === "childInitialized" ) {
+        assertions.send('result', 'success');
+      }
+    });
+
+    document.body.appendChild(iframe);
+  });
+}, Oasis.RSVP.rethrow);
```
Comment (hjdivad) [inline test/fixtures/same_origin_parent.js:32]: just FYI, you don't want `rethrow` as the sibling failure handler, you want it as a downstream failure handler or it won't deal with errors thrown in the fulfilment handler.

`then(doSomething).fail(RSVP.rethrow)`

Comment (stefanpenner) [inline test/fixtures/same_origin_parent.js:32]: https://gist.github.com/stefanpenner/587e5f047d2f412fe463

Comment (cyril-sf) [inline test/fixtures/same_origin_parent.js:32]: Good catch.

**Pull request merged**